src/App.tsx

```diff
@@ -1,12 +1,96 @@
 import React from 'react';
+import 'bulma';
+import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
 
 // import { getAll, get5First, getRed } from './api/goods';
 // or
 // import * as goodsAPI from './api/goods';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+import { getAll } from './api/goods';
+import { GoodsList } from './components/GoodsList';
+
+type Props = {};
+type State = {
+  goods: Good[] | [];
+};
+
+class App extends React.Component<Props, State> {
+  state: State = {
+    goods: [],
+  };
+
+  handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+
+    if (event.currentTarget.name === 'allGoods') {
+      getAll()
+        .then((goods: Good[]) => {
+          this.setState({ goods });
+        });
+    }
+
+    if (event.currentTarget.name === 'firstFive') {
+      getAll()
+        .then((goods: Good[]) => (
+          goods.filter((_, i) => i < 5)
+        ))
+        .then((x: Good[]) => {
+          this.setState({ goods: x });
+        });
+    }
```

Comment:
By condition, you should sort users, after that return first five

Reply:
fixed, thanks